test/framework/src/main/java/org/elasticsearch/test/AbstractSerializingTestCase.java

```diff
@@ -34,9 +35,17 @@
      * both for equality and asserts equality on the two instances.
      */
     public final void testFromXContent() throws IOException {
-        AbstractXContentTestCase.testFromXContent(NUMBER_OF_TEST_RUNS, this::createTestInstance, supportsUnknownFields(),
-                getShuffleFieldsExceptions(), getRandomFieldsExcludeFilter(), this::createParser, this::doParseInstance,
-                this::assertEqualInstances, true, getToXContentParams());
+        AbstractXContentTestCase.testFromXContent(
+                NUMBER_OF_TEST_RUNS,
+                this::createTestInstance,
+                supportsUnknownFields(),
+                getShuffleFieldsExceptions(),
+                getRandomFieldsExcludeFilter(),
+                this::createParser,
+                this::doParseInstance,
+                this::assertEqualInstances,
+                assertToXContentEquivalence(),
```

Comment:
This is the key change, from `true` to use the `assertToXContentEquivalence` hook. This is why I like long parameter lists on a single line. 😇

Were this method already formatted on a single line per parameter, the diff would have made this change obvious:

```
diff --git a/test/framework/src/main/java/org/elasticsearch/test/AbstractSerializingTestCase.java b/test/framework/src/main/java/org/elasticsearch/test/AbstractSerializingTestCase.java
index 7e945a8414c..5aeb30bfdbd 100644
--- a/test/framework/src/main/java/org/elasticsearch/test/AbstractSerializingTestCase.java
+++ b/test/framework/src/main/java/org/elasticsearch/test/AbstractSerializingTestCase.java
@@ -44,7 +44,7 @@ public abstract class AbstractSerializingTestCase<T extends ToXContent & Writeab
                 this::createParser,
                 this::doParseInstance,
                 this::assertEqualInstances,
-                true,
+                assertToXContentEquivalence(),
                 getToXContentParams());
     }
 
```

Reply:
Good point!